### ISSUE TITLE
sponge error

### ISSUE THREAD

**Issue by axin0452** [BeardlessBrady/Currency-Mod#76]
[14:12:52] [Server thread/ERROR] [FML]: Parsing error loading built-in advancement modcurrency:recipes/onbillrecieved
com.google.gson.JsonSyntaxException: Unknown recipe 'modcurrency:atm'
        at net.minecraft.advancements.AdvancementRewards$Deserializer.deserialize(AdvancementRewards.java:171) ~[l$a.class:?]
        at net.minecraft.advancements.AdvancementRewards$Deserializer.deserialize(AdvancementRewards.java:147) ~[l$a.class:?]
        at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[TreeTypeAdapter.class:?]
        at com.google.gson.Gson.fromJson(Gson.java:887) ~[Gson.class:?]
        at com.google.gson.Gson.fromJson(Gson.java:952) ~[Gson.class:?]
        at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:162) ~[TreeTypeAdapter$GsonContextImpl.class:?]
        at net.minecraft.util.JsonUtils.func_188179_a(SourceFile:439) ~[rc.class:?]
        at net.minecraft.util.JsonUtils.func_188177_a(SourceFile:455) ~[rc.class:?]
        at net.minecraft.advancements.Advancement$Builder.func_192059_a(SourceFile:203) ~[i$a.class:?]
        at net.minecraft.advancements.AdvancementManager$1.deserialize(AdvancementManager.java:50) ~[ns$1.class:?]
        at net.minecraft.advancements.AdvancementManager$1.deserialize(AdvancementManager.java:46) ~[ns$1.class:?]
        at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[TreeTypeAdapter.class:?]
        at net.minecraft.util.JsonUtils.func_188173_a(SourceFile:492) ~[rc.class:?]
        at net.minecraft.util.JsonUtils.func_193839_a(SourceFile:532) ~[rc.class:?]
        at net.minecraftforge.common.ForgeHooks.lambda$loadAdvancements$0(ForgeHooks.java:1357) ~[ForgeHooks.class:?]
        at net.minecraftforge.common.crafting.CraftingHelper.findFiles(CraftingHelper.java:821) [CraftingHelper.class:?]
        at net.minecraftforge.common.ForgeHooks.loadAdvancements(ForgeHooks.java:1339) [ForgeHooks.class:?]
        at net.minecraftforge.common.ForgeHooks.loadAdvancements(ForgeHooks.java:1313) [ForgeHooks.class:?]
        at net.minecraft.advancements.AdvancementManager.func_192779_a(AdvancementManager.java:69) [ns.class:?]
        at net.minecraft.advancements.AdvancementManager.<init>(AdvancementManager.java:60) [ns.class:?]
        at net.minecraft.world.WorldServer.func_175643_b(WorldServer.java:156) [oo.class:?]
        at org.spongepowered.common.world.WorldManager.createWorldFromProperties(WorldManager.java:803) [WorldManager.class:1.12.2-2705-7.1.0-BETA-3481]
        at org.spongepowered.common.world.WorldManager.loadAllWorlds(WorldManager.java:771) [WorldManager.class:1.12.2-2705-7.1.0-BETA-3481]
        at net.minecraft.server.MinecraftServer.func_71247_a(MinecraftServer.java:3545) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:270) [nz.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:486) [MinecraftServer.class:?]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_131]

**Comment by Beardlessbrady** [comment]
Maybe explain more than just dropping a crash log

**Comment by axin0452** [comment]
This mod is great, and it's also important to me.so,What information do I need to provide?

**Comment by Beardlessbrady** [comment]
Not sure why you closed the issue...Anywho, firstly I need to know if you are able to reproduce it. If so I need to know in what conditions it occurs (Which version of forge, the mod, if any other mods are present)

**Comment by huntersan9** [comment]
Having the same issue with a fresh setup of forge.

First Error in the log:
`[Server thread/ERROR] [FML]: Parsing error loading recipe modcurrency:atm
com.google.gson.JsonSyntaxException: Unknown item 'modcurrency:blockatm'
        at net.minecraftforge.common.crafting.CraftingHelper.getItemStack(CraftingHelper.java:212) ~[CraftingHelper.class:?]
        at net.minecraftforge.common.crafting.CraftingHelper.lambda$init$14(CraftingHelper.java:516) ~[CraftingHelper.class:?]
        at net.minecraftforge.common.crafting.CraftingHelper.getRecipe(CraftingHelper.java:409) ~[CraftingHelper.class:?]
        at net.minecraftforge.common.crafting.CraftingHelper.lambda$loadRecipes$22(CraftingHelper.java:711) ~[CraftingHelper.class:?]
        at net.minecraftforge.common.crafting.CraftingHelper.findFiles(CraftingHelper.java:822) ~[CraftingHelper.class:?]
        at net.minecraftforge.common.crafting.CraftingHelper.loadRecipes(CraftingHelper.java:668) ~[CraftingHelper.class:?]
        at java.util.ArrayList.forEach(ArrayList.java:1257) [?:1.8.0_172]
        at net.minecraftforge.common.crafting.CraftingHelper.loadRecipes(CraftingHelper.java:620) [CraftingHelper.class:?]
        at net.minecraftforge.fml.common.Loader.initializeMods(Loader.java:742) [Loader.class:?]
        at net.minecraftforge.fml.server.FMLServerHandler.finishServerLoading(FMLServerHandler.java:108) [FMLServerHandler.class:?]
        at net.minecraftforge.fml.common.FMLCommonHandler.onServerStarted(FMLCommonHandler.java:338) [FMLCommonHandler.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:219) [nz.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:486) [MinecraftServer.class:?]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_172]
`

And then (likely caused by the first) 
`[Server thread/ERROR] [FML]: Parsing error loading built-in advancement modcurrency:recipes/onbillrecieved
com.google.gson.JsonSyntaxException: Unknown recipe 'modcurrency:atm'
        at net.minecraft.advancements.AdvancementRewards$Deserializer.deserialize(AdvancementRewards.java:171) ~[l$a.class:?]
        at net.minecraft.advancements.AdvancementRewards$Deserializer.deserialize(AdvancementRewards.java:147) ~[l$a.class:?]
        at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[TreeTypeAdapter.class:?]
        at com.google.gson.Gson.fromJson(Gson.java:887) ~[Gson.class:?]
        at com.google.gson.Gson.fromJson(Gson.java:952) ~[Gson.class:?]
        at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:162) ~[TreeTypeAdapter$GsonContextImpl.class:?]
        at net.minecraft.util.JsonUtils.func_188179_a(SourceFile:439) ~[rc.class:?]
        at net.minecraft.util.JsonUtils.func_188177_a(SourceFile:455) ~[rc.class:?]
        at net.minecraft.advancements.Advancement$Builder.func_192059_a(SourceFile:203) ~[i$a.class:?]
        at net.minecraft.advancements.AdvancementManager$1.deserialize(AdvancementManager.java:50) ~[ns$1.class:?]
        at net.minecraft.advancements.AdvancementManager$1.deserialize(AdvancementManager.java:46) ~[ns$1.class:?]
        at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[TreeTypeAdapter.class:?]
        at net.minecraft.util.JsonUtils.func_188173_a(SourceFile:492) ~[rc.class:?]
        at net.minecraft.util.JsonUtils.func_193839_a(SourceFile:532) ~[rc.class:?]
        at net.minecraftforge.common.ForgeHooks.lambda$loadAdvancements$0(ForgeHooks.java:1356) ~[ForgeHooks.class:?]
        at net.minecraftforge.common.crafting.CraftingHelper.findFiles(CraftingHelper.java:822) [CraftingHelper.class:?]
        at net.minecraftforge.common.ForgeHooks.loadAdvancements(ForgeHooks.java:1338) [ForgeHooks.class:?]
        at net.minecraftforge.common.ForgeHooks.loadAdvancements(ForgeHooks.java:1312) [ForgeHooks.class:?]
        at net.minecraft.advancements.AdvancementManager.func_192779_a(AdvancementManager.java:69) [ns.class:?]
        at net.minecraft.advancements.AdvancementManager.<init>(AdvancementManager.java:60) [ns.class:?]
        at net.minecraft.world.WorldServer.func_175643_b(WorldServer.java:156) [oo.class:?]
        at net.minecraft.server.MinecraftServer.func_71247_a(MinecraftServer.java:298) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:270) [nz.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:486) [MinecraftServer.class:?]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_172]
`

Running:
Forge 1.12.2-14.23.5.2768
Currency-1.12.0-1.3.6
Java 1.8.0_172

Fresh install.
Hope that helps.

**Comment by Beardlessbrady** [comment]
I will look into it when I can. Thanks

**Comment by huntersan9** [comment]
I've since looked into how to make mods, or at least try to find out why I get errors.  Thought I'd lend a hand.

It looks like you had disabled the ATM as noted in the TODO.txt, 

> Removed ATM as it needs work (Will be re added in mod rewrite)


When it looked at the recipe for the atm, since it didn't load the atm, it didn't know what the block was, so error.
     - `Removed the file assets/modcurrency/recipes/atm.json`
It also errored when it tried to find that recipe for the advancement, onbillrecieved
     - `Removed line 9 from assets/modcurrency/advancements/recipes/onbillrecieved.json`

Tested with same setup and loads fine.

It's a simple kludge but it works.  Hope that helps you out!  Good luck with the new version.
